packages/drivers/odsp-driver/src/odspDocumentStorageManager.ts

```diff
@@ -214,12 +215,205 @@ export class OdspDocumentStorageService extends OdspDocumentStorageServiceBase {
 	}
 
 	public async getSnapshot(snapshotFetchOptions?: ISnapshotFetchOptions): Promise<ISnapshot> {
-		assert(this.hostPolicy.supportGetSnapshotApi !== true, "api should not be called yet");
-		// This is just temporary as this api is not yet enabled in service policies.
-		return this.fetchSnapshot(
-			this.hostPolicy.snapshotOptions,
-			snapshotFetchOptions?.scenarioName,
+		// If no groupId is specified, then fetch the snapshot with "default" groupId.
+		const groupIds = snapshotFetchOptions?.groupIds ?? [];
+		if (groupIds.length === 0) {
+			groupIds.push(defaultGroupIdForSnapshot);
```

Comment:
Can this (leveraging defaultGroupIdForSnapshot) be done at the lowest layer? I.e. right before doing fetch()?
This way all the layers can assume that empty list means - initial snapshot.

Reply:
That said, see comment in prefetchLatestSnapshot() - I think there is a bug when handling []

Reply:
Moved it to lowest layer just before fetch.